extensions/amp-position-observer/amp-position-observer.md

```diff
@@ -0,0 +1,220 @@
+
+## Example
+Consider the same clock animation, but this time the hand animates with time, except
+we like the animation to start when clock is at least 50% visible and pause as soon
+as clock becomes less than 50% visible.
+
+![visibility demo](https://user-images.githubusercontent.com/2099009/29105727-a7d9a80a-7c84-11e7-8d4a-794f38ea5a5c.gif)
+
+
+```html
+<!-- An animation that rotates a clock hand 180 degrees. -->
+<!--
+   Note that we are NOT setting `trigger=visibility`
+   since we will manually trigger the animation
+-->
+<!--
+   Also note that this is the same animation as the scroll-bound version above
+   the animation is the same, just the triggering mechanism with
+   `amp-position-observer` is different!
+-->
+<amp-animation id="clockAnim" layout="nodisplay">
+  <script type="application/json">
+    {
+    "duration": "3s",
+    "fill": "both",
+    "direction": "alternate",
+    "animations": [
+      {
+        "selector": "#clock-scene .clock-hand",
+        "keyframes": [
+          { "transform": "rotate(-180deg)" },
+          { "transform": "rotate(0deg)" }
+        ]
+      }
+    ]
+  }
+  </script>
+</amp-animation>
+
+<!-- The clock container -->
+<div id="clock-scene">
+  <amp-img layout="responsive" width=2 height=1.5 src="./img/clock.jpg">
+    <div class="clock-hand"></div>
+  </amp-img>
+</div>
+
+<!--
+  Use amp-position-observer to tie the start/pause of the animation with
+  the visibility of the scene.
+-->
+
+<amp-position-observer
+  target-selector="#clock-scene"
+  intersection-ratios="0.5"
+  on="enter:clockAnim.start;exit:clockAnim.pause"
+  layout="nodisplay">
+</amp-position-observer>
+```
+
+## Attributes
+
+#### target-selector (optional)
+Specifies what element to observe.
+If **not specified** the **parent** of `<amp-position-observer>` will be used as the target.
+
+#### intersection-ratios (optional)
+
+Defines how much of the target should be visible in the viewport before `<amp-position-observer>`
+triggers any of its events. The value can a number between 0 and 1.  The default value is 0.
+
+Different ratios for top vs. bottom can be specified by providing two values (`<top> <bottom>`).
+
+* `intersection-ratios="0"`: Indicates that `enter` is triggered as soon as a single pixel
+of the target comes into viewport and `exit` is triggered as soon as the very last pixel
+of the target goes out of the viewport.
+* `intersection-ratios="0.5"`: Indicates that `enter` is triggered as soon as 50% of
+of the target comes into viewport and `exit` is triggered as soon as less than
+50% of the target is in the viewport.
+* `intersection-ratios="1"`: Indicates that `enter` is triggered when target is fully visible
+and `exit` is triggered as soon as a single pixel goes out of the viewport.
+* `intersection-ratios="0 1"`: Makes the conditions different depending on whether
+the target is entering/exiting from top (0 will be used) or bottom (1 will be used).
+
+
+#### exclusion-margins (optional)
```

Comment:
A bit bikeshed on name. Would it be better to rename to root margins or viewport margins to get it closer to InOb terminology which we use elsewhere?